backend/src/main/java/ai/verta/modeldb/cron_jobs/DeleteEntitiesCron.java

```diff
@@ -0,0 +1,666 @@
+                  .equals(ProjectVisibility.ORG_SCOPED_PUBLIC),
+              "_GLOBAL_SHARING");
+      roleBindingNames.addAll(workspaceRoleBindingNames);
+    }
+
+    roleService.deleteAllResources(
+        allowedProjects.stream().map(ProjectEntity::getId).collect(Collectors.toList()),
+        ModelResourceEnum.ModelDBServiceResourceTypes.PROJECT);
+  }
+
+  private void deleteExperiments(Session session) {
+    LOGGER.debug("Experiment deleting");
+    String deleteExperimentQueryString =
+        new StringBuilder("FROM ")
+            .append(ExperimentEntity.class.getSimpleName())
+            .append(" ex WHERE ex.")
+            .append(ModelDBConstants.DELETED)
+            .append(" = :deleted ")
+            .toString();
+
+    Query experimentDeleteQuery = session.createQuery(deleteExperimentQueryString);
+    experimentDeleteQuery.setParameter("deleted", true);
+    experimentDeleteQuery.setMaxResults(this.recordUpdateLimit);
+    LOGGER.debug("Experiment delete query: {}", experimentDeleteQuery.getQueryString());
+    List<ExperimentEntity> experimentEntities = experimentDeleteQuery.list();
+
+    List<String> experimentIds = new ArrayList<>();
+    if (!experimentEntities.isEmpty()) {
+      List<String> roleBindingNames = new LinkedList<>();
+      for (ExperimentEntity experimentEntity : experimentEntities) {
+        experimentIds.add(experimentEntity.getId());
+        String ownerRoleBindingName =
+            roleService.buildRoleBindingName(
+                ModelDBConstants.ROLE_EXPERIMENT_OWNER,
+                experimentEntity.getId(),
+                experimentEntity.getOwner(),
+                ModelResourceEnum.ModelDBServiceResourceTypes.EXPERIMENT.name());
+        if (ownerRoleBindingName != null) {
+          roleBindingNames.add(ownerRoleBindingName);
+        }
+      }
+
+      if (!roleBindingNames.isEmpty()) {
+        roleService.deleteRoleBindings(roleBindingNames);
+      }
+
+      Transaction transaction = session.beginTransaction();
+      String updateDeletedStatusExperimentRunQueryString =
+          new StringBuilder("UPDATE ")
+              .append(ExperimentRunEntity.class.getSimpleName())
+              .append(" expr ")
+              .append("SET expr.")
+              .append(ModelDBConstants.DELETED)
+              .append(" = :deleted ")
+              .append(" WHERE expr.")
+              .append(ModelDBConstants.EXPERIMENT_ID)
+              .append(" IN (:experimentIds)")
+              .toString();
+      Query deletedExperimentRunQuery =
+          session.createQuery(updateDeletedStatusExperimentRunQueryString);
+      deletedExperimentRunQuery.setParameter("deleted", true);
+      deletedExperimentRunQuery.setParameter("experimentIds", experimentIds);
+      deletedExperimentRunQuery.executeUpdate();
+
+      for (ExperimentEntity experimentEntity : experimentEntities) {
+        session.delete(experimentEntity);
+      }
```

Comment:
Same comment as for projects.

Reply:
Comment: https://github.com/VertaAI/modeldb/pull/651#discussion_r429177879